common/src/main/java/org/corfudb/common/metrics/micrometer/MicroMeterUtils.java

```diff
@@ -70,6 +136,41 @@ public static void measure(double measuredValue, String name, String... tags) {
         summary.ifPresent(s -> s.record(measuredValue));
     }
 
+    private static List<Tag> toTagIterable(String... tags) {
+        if (tags.length % 2 != 0) {
+            throw new IllegalArgumentException("Only key-value pairs allowed.");
+        }
+        if (tags.length == 0) {
+            return ImmutableList.of();
+        }
+        return IntStream.range(1, tags.length)
+                .filter(i -> i % 2 != 0)
```

Comment:
I don't understand, why are we filtering based on even/odd?

Reply:
The gauge api accepts only the iterable of tags but the input is the array of strings. This function basically says "starting from index 1 get every even-indexed entry and couple it with a previous entry into a tag".

Reply:
In that case, zipping two streams would be more readable. I believe Guava provides `Stream<T>.zip(...)`.